08_project/willshen8/pkg/puppy/types.go

```diff
@@ -0,0 +1,8 @@
+package puppy
+
+type Puppy struct {
+	ID    uint32
+	Breed string
+	Color string
+	Value string
```

Comment:
That's an interesting choice of data type for a value - may I ask why?

Reply:
@alextmz. I set the value to `string` as originally I wanted to sell the puppy to overseas, so the value can be multi-currency. LOL. (seriously though, this was my original thought, as the value should have the currency sign - e.g. $. But there might be a better way of dealing with it in real applictions?